File: Recien traducido/DP/BurstBalloonMaximizeCoins.py
def getMax(A): 
    N = len(A) 
    A = [1] + A + [1]
    dp = [[0 for x in range(N + 2)] for y in range(N + 2)] 
      
    for length in range(1, N + 1): 
        for left in range(1, N-length + 2): 
            right = left + length -1
            for last in range(left, right + 1): 
                dp[left][right] = max(dp[left][right], \ 
                                      dp[left][last-1] + \ 
                                      A[left-1] * A[last] * A[right + 1] + \ 
                                      dp[last + 1][right]) 
    return(dp[1][N]) 
  
A = [1, 2, 3, 4, 5] 
print(getMax(A)) 

File: Recien traducido/DP/BurstBalloonMaximizeCoins.cpp

#include <bits/stdc++.h> 
#include <iostream> 
using namespace std;

int getMax(int A[], int N) {
    int B[N + 2];
    B[0] = 1;
    B[N + 1] = 1;
    for (int i = 1; i <= N; i++)
        B[i] = A[i - 1];
    int dp[N + 2][N + 2];
    memset(dp, 0, sizeof (dp));

    for (int length = 1; length < N + 1; length++) {
        for (int left = 1; left < N - length + 2; left++) {
            int right = left + length - 1;
            for (int last = left; last < right + 1; last++) {
                dp[left][right] = max(dp[left][right],
                        dp[left][last - 1] +
                        B[left - 1] * B[last] * B[right + 1] +
                        dp[last + 1][right]);
            }
        }
    }
    return dp[1][N];
}

int main() {
    int A[] = {1, 2, 3, 4, 5};
    int N = sizeof (A) / sizeof (A[0]);
    cout << getMax(A, N) << endl;
}
